libretroshare/src/services/p3msgservice.cc

```diff
@@ -1954,17 +2086,31 @@ void p3MsgService::notifyDataStatus( const GRouterMsgPropagationId& id,
 			return;
 		}
 
+#if 0
 		delete it2->second;
 		msgOutgoing.erase(it2);
+#else
+		// Do not delete it move to sent folder instead!
+		it2->second->msgFlags &= ~RS_MSG_FLAGS_PENDING;
+		imsg[msg_id] = it2->second;
+		msgOutgoing.erase(it2);
+#endif
 
 		RsServer::notify()->notifyListChange( NOTIFY_LIST_MESSAGELIST,
 		                                      NOTIFY_TYPE_ADD );
 		IndicateConfigChanged();
 
+		using Evt_t = RsMailStatusEvent;
```

Comment:
I'm wondering if there is a way to put these 4 lines into one, maybe using a method like sendEvent(RsMailStatusEvent(...));.

Reply:
A macro could be used for this, but I would keep that for another PR